### DISCUSSION TITLE
Socket Electricity Calibration

### DISCUSSION THREAD

**Discussion by YuriiMaiboroda** [make-all/tuya-local#467] [General]
My smart socket (Elivko smart plug 16A) does not correctly show the mains voltage (show about 10 volts more than other devices).
Has anyone encountered such a problem?  How can I calibrate the socket?

I heard that Tasmota has a calibration function. But I'm not ready to reflash the socket yet. I just started to dive into the world of smart things.

**Comment by make-all** [comment]
Some devices expose calibration DPS, but I don't know  if they are writable (maybe there is a way to unlock them, the test bit may be involved). Even those that do not output those DPS may accept writing them if they are based on the standard Tuya smartplug firmware with the standard DPS layout. There may be some documentation on the Tuya developer site aimed at manufacturers.

**Comment by YuriiMaiboroda** [comment]
As far as I understood, reading articles for manufacturers, calibration is possible only at the production stage. Ordinary users are not provided with the possibility of calibration. The device must be sold already calibrated.

**Reply by make-all** [reply]
That is why I think the test bit field might have something to do with it.  The device needs to be set into an "under test" state before it will accept new calibration values.  It may be a door knock protocol though, rather than a simple set command.
